README.md

```diff
@@ -0,0 +1,362 @@
+    end
+  
+    def self.silly_decrypt(options)
+      options[:value].reverse.gsub(/#{options[:secret_key]}$/, '')
+    end
+  end
+```
+
+Simply set up your class like so:
+
+```ruby
+  class User
+    attr_encrypted :email, secret_key: 'a secret key', encryptor: SillyEncryptor, encrypt_method: :silly_encrypt, decrypt_method: :silly_decrypt
+  end
+```
+
+Any options that you pass to `attr_encrypted` will be passed to the encryptor class along with the `:value` option which contains the string to encrypt/decrypt. Notice that the above example uses `:secret_key` instead of `:key`. See [encryptor](https://github.com/attr-encrypted/encryptor) for more info regarding the default encryptor class.
+
+
+### The `:algorithm` option
+
+The default `Encryptor` class uses the standard ruby OpenSSL library. Its default algorithm is `aes-256-gcm`. You can modify this by passing the `:algorithm` option to the `attr_encrypted` call like so:
+
+```ruby
+  class User
+    attr_encrypted :email, key: 'a secret key', algorithm: 'aes-256-cbc'
+  end
+```
+
+To view a list of all cipher algorithms that are supported on your platform, run the following code in your favorite Ruby REPL:
+
+```ruby
+  require 'openssl'
+  puts OpenSSL::Cipher.ciphers
+```
+See [Encryptor](https://github.com/attr-encrypted/encryptor#algorithms) for more information.
+
+
+### The `:encode`, `:encode_iv`, `:encode_salt`, and `:default_encoding` options
+
+You're probably going to be storing your encrypted attributes somehow (e.g. filesystem, database, etc). You can simply pass the `:encode` option to automatically encode/decode when encrypting/decrypting. The default behavior assumes that you're using a string column type and will base64 encode your cipher text. If you choose to use the binary column type then encoding is not required, but be sure to pass in `false` with the `:encode` option.
+
+```ruby
+  class User
+    attr_encrypted :email, key: 'some secret key', encode: true, encode_iv: true, encode_salt: true
+  end
+```
+
+The default encoding is `m` (base64). You can change this by setting `encode: 'some encoding'`. See [`Arrary#pack`](http://ruby-doc.org/core-2.3.0/Array.html#method-i-pack) for more encoding options.
+
+
+### The `:marshal`, `:dump_method`, and `:load_method` options
+
+You may want to encrypt objects other than strings (e.g. hashes, arrays, etc). If this is the case, simply pass the `:marshal` option to automatically marshal when encrypting/decrypting.
+
+```ruby
+  class User
+    attr_encrypted :credentials, key: 'some secret key', marshal: true
+  end
+```
+
+You may also optionally specify `:marshaler`, `:dump_method`, and `:load_method` if you want to use something other than the default `Marshal` object.
+
+
+## ORMs
+
+### ActiveRecord
+
+If you're using this gem with `ActiveRecord`, you get a few extra features:
+
+#### Default options
+
+The `:encode` option is set to true by default.
+
+#### Dynamic `find_by_` and `scoped_by_` methods
```

Comment:
I suppose that it doesn't hurt to mention it here. I was trying to be as terse as possible without missing anything. On a related note, perhaps I should mention deprecating `:single_iv_and_salt` mode. I think in the next major release we should remove it.